pkg/chunk/purger/purger.go

```diff
@@ -454,10 +444,7 @@ func (p *Purger) pullDeleteRequestsToPlanDeletes() error {
 			return err
 		}
 
-		p.inProcessRequestIDsMtx.Lock()
-		p.inProcessRequests[deleteRequest.UserID] = deleteRequest
-		p.inProcessRequestIDsMtx.Unlock()
-
+		p.inProcessRequests.add(deleteRequest.UserID, &deleteRequest)
```

Comment:
This takes address of variable declared in `for` loop. This address will be the same for all iterations.

For this to work correctly, copy of deleteRequest must be made:

```suggestion
		deleteRequestCopy := deleteRequest
		p.inProcessRequests.add(deleteRequest.UserID, &deleteRequestCopy)
```

Reply:
Thanks for the catch! I totally missed it.